platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/DataFileMetadata.java

```diff
@@ -154,60 +142,35 @@ public static DataFileMetadata readFromFile(Path file) throws IOException {
             throw new IllegalArgumentException("No metadata found in file: " + file);
         }
 
-        return new DataFileMetadata(
-                itemsCount, index, Instant.ofEpochSecond(creationSeconds, creationNanos), compactionLevel);
+        return new DataFileMetadata(index, Instant.ofEpochSecond(creationSeconds, creationNanos), compactionLevel);
     }
 
-    void writeTo(final BufferedData out) {
+    <T extends WritableSequentialData> void writeTo(final T out) {
```

Comment:
I do like changing this method to work with an interface type (WritableSequentialData instead of BufferedData), but why does it have to use generics? Would the following work instead?

```java
void writeTo(final WritableSequentialData out) { ... }
```